app/models/ext_management_system.rb

```diff
@@ -503,4 +503,23 @@ def blacklisted_event_names
   def self.blacklisted_events
     BlacklistedEvent.where(:provider_model => name, :ems_id => nil)
   end
+
+  def self.credential_validation(ems, auth_type, creds, revalidate)
```

Comment:
Spoke with @agrare, and part of the problem is that the EMS might not exist yet.  I don't really like that it serializes the ems object on the queue, but I think it's ok for now.